src/main/java/uk/gov/ons/census/action/messaging/EventReceiver.java

```diff
@@ -76,18 +85,66 @@ private void processCaseCreatedEvent(CollectionCase collectionCase) {
     newCase.setFieldCoordinatorId(collectionCase.getFieldCoordinatorId());
     newCase.setFieldOfficerId(collectionCase.getFieldOfficerId());
     newCase.setCeExpectedCapacity(collectionCase.getCeExpectedCapacity());
+    newCase.setReceiptReceived(false);
 
     caseRepository.save(newCase);
   }
 
-  private void processCaseUpdatedEvent(CollectionCase collectionCase) {}
+  private void processCaseUpdatedEvent(Event event, CollectionCase collectionCase) {
+    String caseId = collectionCase.getId();
+
+    Optional<Case> cazeOpt = caseRepository.findByCaseId(UUID.fromString(caseId));
+
+    if (cazeOpt.isEmpty()) {
+      log.error(String.format(CASE_NOT_FOUND_ERROR, caseId));
+      throw new RuntimeException();
+    }
+
+    Case updatedCase = cazeOpt.get();
```

Comment:
95% of this code can be re-used from lines 54 through 88 above

Reply:
Extracted to private method.